app/controllers/ticket_purchases_controller.rb

```diff
@@ -4,11 +4,11 @@ class TicketPurchasesController < ApplicationController
   authorize_resource :conference_registrations, class: Registration
 
   def create
+    TicketPurchase.destroy_all(user_id: current_user.id, conference_id: @conference.id, paid: false)
     message = TicketPurchase.purchase(@conference, current_user, params[:tickets][0])
     if message.blank?
-      if current_user.ticket_purchases.any?
-        redirect_to conference_conference_registrations_path(@conference.short_title),
-                    notice: "Thank you for supporting #{@conference.title} by purchasing a ticket."
+      if current_user.ticket_purchases.where(paid: false).any?
```

Comment:
I can't reuse the unpaid function here because it is a ticket-model specific function... so I'll have to either define the function again or just use a where clause.